### PR TITLE
Ported to Hapi v17

### LINKED ISSUE #24: Support hapi v17
Hi there!

Because one of my plugins is relying on `hapi-pino` and I use it in several projects with the latest stuff, I was wondering if you guys already work on supporting the `v17.0.0` of `hapi` which seems to be released at 1st of November :)

Thanks and best.

### PR DESCRIPTION
**Pull request by mcollina**
Hurray on the big perf improvement! Good was removed because it was not
ported yet.

@felixheck @timcosta please test and review!

Fixes #24

### CODE REVIEW

**Comment by felixheck** [comment]
Changes LGTM as well.
I'll test it tomorrow.

**Comment by timcosta** [comment]
LGTM @mcollina, thanks!!

**Comment by felixheck** [comment]
Why do you filter `request` events so that only specific errors get logged?

> The 'request' event type emits internal request events generated by the framework as well as application events logged with request.log().

So all `request.log()` gets ignored, aren't there?

File: index.js [hunk 1]
```diff
@@ -52,26 +52,23 @@ function register (server, options, next) {
...
   // set a logger for each request
-  server.ext('onRequest', (request, reply) => {
+  server.ext('onRequest', (request, h) => {
     request.logger = logger.child({ req: request })
-    reply.continue()
+    return h.continue
   })
 
-  server.on('log', function (event) {
+  server.events.on('log', function (event, tags) {
```
Comment (felixheck) [inline index.js]: what's that `tags` arg?

File: index.js [hunk 2]
```diff
@@ -3,22 +3,22 @@
...
+const levelTags = {
   trace: 'trace',
   debug: 'debug',
   info: 'info',
   warn: 'warn',
   error: 'error'
 }
 
-function register (server, options, next) {
+async function register (server, options, nex) {
```
Comment (felixheck) [inline index.js]: what's that `nex` arg?

File: index.js [hunk 3]
```diff
@@ -84,21 +81,24 @@ function register (server, options, next) {
...
   })
 
-  tryAddEvent(server, options, 'ext', 'onPostStart', function (s, cb) {
+  tryAddEvent(server, options, 'ext', 'onPostStart', async function (s) {
     logger.info(server.info, 'server started')
-    cb()
   })
 
-  tryAddEvent(server, options, 'ext', 'onPostStop', function (s, cb) {
+  tryAddEvent(server, options, 'ext', 'onPostStop', async function (s, cb) {
```
Comment (felixheck) [inline index.js]: `cb` is unneeded

**Comment by mcollina** [comment]
@felixheck I didn't know how to port https://github.com/pinojs/hapi-pino/blob/master/index.js#L69-L85 to Hapi v17. What I've done passes the same tests that were passing before. However, I do not claim it's correct. I couldn't find good docs/explanation on how to do what I wanted, e.g. "get all errors once", because I started to get double logged events.

**Comment by mcollina** [comment]
> So all request.log() get ignored, aren't they?

No, `request.log()` should pass through, tests covers that.

**Comment by felixheck** [comment]
When I use `request.log()` in a minimal working example it does not get logged..
When I let the request fail, the error gets logged.

**Comment by mcollina** [comment]
@felixheck can you paste a simple example to reproduce or send a PR to this one with a failing test? Unfortunately this is as close to a full rewrite as possible.

**Comment by felixheck** [comment]
As far as I understood the events `request-error` and `request` just got merged into a single one.
So maybe it's enough to handle both cases in the event handler?

```js
const hapi = require('hapi')
const hapiPino = require('hapi-pino')

async function init() {
  const server = hapi.server({ port: 1337 })
  
  await server.register({
    plugin: hapiPino,
    options: {
      logEvents: false
    }
  })

  server.route({
    method: 'GET',
    path: '/',
    handler(req) {
      req.log('foobar')
      server.log('barfoo')
      return 'Hello World'
    }
  })

  await server.start()
}

init()
```

**Comment by mcollina** [comment]
@felixheck please check now.

**Comment by felixheck** [comment]
Looks much better but currently not able to test it in combination with my plugin based on hapi-pino, because almost all of my tests seem to be broken after updating hapi -.-

But first tests with my demo looks good.

**Comment by mcollina** [comment]
This is close to a full rewrite. Hapi v17 is a completely different framework.

**Comment by StianOvrevage** [comment]
I cannot find pino3 on npm. I also notice you deleted the v17 branch. Does this mean hapi-pino will not be ported to v17?

**Comment by StianOvrevage** [comment]
Turns out I'm a n00b at npm apparently . Colin on Slack tipped me of `npm i hapi-pino@3` which looks good so far :-)

**Pull request merged**